### DISCUSSION TITLE
How to pass redoc options

### DISCUSSION THREAD

**Discussion by benfoster** [rohit-gohri/redocusaurus#229] [Q&A] [answered]
Redoc has a number of configurable options documented [here](https://github.com/Redocly/redoc/#redoc-options-object). How do you pass these to the Redocusaurus plugin?

I saw one issue that suggested to add this to the `theme` object but this does not work. For example:

```
    // Redocusaurus config
    [
      'redocusaurus',
      {
        // Plugin Options for loading OpenAPI files
        specs: [
          {
            spec: 'openapi/openapi.yaml',
            route: '/api/',
            config: {
              disableSearch: true
            }
          },
        ],
        // Theme Options for modifying how redoc renders them
        theme: {
          // Change with your site colors
          primaryColor: '#1890ff',
          redocOptions: {
            disableSearch: true,
            hideDownloadButton: true
          }
        }
      },
    ],
```

**Comment by rohit-gohri** [comment, marked as answer]
You can pass it as `options` - https://redocusaurus.vercel.app/docs/getting-started/theme-options

**Reply by benfoster** [reply]
Do you happen to have a working example? I have tried a variety of options and they don't appear to have any impact:

```
    [
      'redocusaurus',
      {
        // Plugin Options for loading OpenAPI files
        specs: [
          {
            spec: 'openapi/openapi.yaml',
            route: '/api/'
          },
        ],
        // Theme Options for modifying how redoc renders them
        theme: {
          // Change with your site colors
          primaryColor: '#1890ff'
        },
        options: {
          disableSearch: true,
          hideDownloadButton: true,
          disableSidebar: true
        }
      },
    ],
```

**Reply by benfoster** [reply]
I figured it out in the end. The `options` parameter should be passed within the `theme` object. I think a sample might help on the docs as the way it is presented currently suggests that `options` and `theme` exist at the same level, which is why I attempted the above.

I was able to find the correct structure [here](https://github.com/rohit-gohri/redocusaurus/blob/main/website/docusaurus.config.js).

**Reply by rohit-gohri** [reply]
It'll be great if you can add your example to the [docs](https://github.com/rohit-gohri/redocusaurus/blob/main/website/docs/getting-started/theme-options.md) and open a PR.
